### ISSUE TITLE
Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job

### ISSUE THREAD

**Issue by afischer-opentext-com** [saltstack/salt#51455]
### Description of Issue/Question

Our Salt Minions (version 2018.3.3) are flood their log file with the following log line

```
2019-01-31 18:59:13,543 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:14,230 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:14,402 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:14,528 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:14,544 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:15,230 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:15,402 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:15,528 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:15,544 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:16,230 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:16,403 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:16,528 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:16,528 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:17,229 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:17,402 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:17,530 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:17,530 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:18,229 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:18,405 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:18,531 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:18,531 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:19,229 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:19,402 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:19,528 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:19,543 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:20,230 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:20,402 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:20,528 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:20,544 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
2019-01-31 18:59:21,230 [salt.minion      :433 ][ERROR   ][4216] Exception 'Schedule' object has no attribute 'loop_interval' occurred in scheduled job
```

We would like to understand what this may mean and how to prevent it. I found so far no ticket describing the issue, only a [gist](https://gist.github.com/PeterS242/bf9bd2af214fe3ed391d510c29f3756a) mentioning the issue as well. Even if this may not be critical, it is super annoying and lets the log size increase to an uncomfortable level. 

### Setup

Our minion configuration is

```
s3.path_style: True
pillarenv_from_saltenv: True
autoload_dynamic_modules: False
open_mode: True
use_superseded: ['module.run']
master: ['salt']
id: minion
ipc_mode: tcp
state_top: salt://top.sls
multiprocessing: False
redis.host: salt-redis
redis.db: '0'
cache.redis.host: salt-redis
# Content of the automatically created _schedule.conf
schedule:
  __mine_interval: {enabled: true, function: mine.update, jid_include: true, maxrunning: 2,
    minutes: 60, return_job: false}
```

### Steps to Reproduce Issue
* Install and run salt-minion on 2013.3.3 py3 salt-minion on Windows
* Inspect the log file

### Versions Report

```
PS C:\salt> cmd /c salt-call.bat --local --versions-report
Salt Version:
           Salt: 2018.3.3

Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: 1.6.1
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: 2.0.5
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.1.6

System Versions:
           dist:
         locale: cp1252
        machine: AMD64
        release: 2008ServerR2
         system: Windows
        version: 2008ServerR2 6.1.7601 SP1 Multiprocessor Free
```

**Comment by dwoz** [comment]
@afischer-opentext-com thanks for reporting this issue

**Comment by twangboy** [comment]
@afischer-opentext-com I can't replicate this on a local box with PY2. I'll try Py3 tomorrow.

**Comment by zbarr** [comment]
Running into the same issue with py3.

Salt Version:
           Salt: 2018.3.3
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: 10.2.1
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.5
      gitpython: 2.1.3
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: 1.6.1
       M2Crypto: Not Installed
           Mako: 1.0.6
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.3 (v3.5.3:1880cb95a742, Jan 16 2017, 16:02:32) [MSC v.1900 64 bit (AMD64)]
   python-gnupg: 0.4.1
         PyYAML: 3.12
          PyZMQ: 16.0.3
           RAET: Not Installed
          smmap: 2.0.5
        timelib: 0.2.4
        Tornado: 4.5.1
            ZMQ: 4.1.6
 
System Versions:
           dist:   
         locale: cp1252
        machine: AMD64
        release: 7
         system: Windows
        version: 7 6.1.7601 SP1 Multiprocessor Free

**Comment by twangboy** [comment]
@zbarr Are you using defaults in the minion config?

**Comment by twangboy** [comment]
Does it start immediately when the minion starts, or do you have to wait a while for it to start dumping messages to the log?

**Comment by zbarr** [comment]
@twangboy I found the issue, I will explain.

I found the following critical error on startup of the salt minion:

2019-02-04 18:08:05,365 [salt.minion      :1014][CRITICAL][4044] Unexpected error while connecting to saltmaster02.concord.global
Traceback (most recent call last):
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 997, in _connect_minion
    yield minion.connect_master(failed=failed)
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1063, in run
    yielded = self.gen.throw(*exc_info)
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1189, in connect_master
    yield self._post_master_init(master)
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1055, in run
    value = future.result()
  File "c:\salt\bin\lib\site-packages\tornado\concurrent.py", line 238, in result
    raise_exc_info(self._exc_info)
  File "<string>", line 4, in raise_exc_info
  File "c:\salt\bin\lib\site-packages\tornado\gen.py", line 1069, in run
    yielded = self.gen.send(value)
  File "c:\salt\bin\lib\site-packages\salt\minion.py", line 1234, in _post_master_init
    cleanup=[master_event(type='alive')])
  File "c:\salt\bin\lib\site-packages\salt\utils\schedule.py", line 88, in __new__
    cls.instance.__singleton_init__(opts, functions, returners, intervals, cleanup, proxy, utils, standalone)
  File "c:\salt\bin\lib\site-packages\salt\utils\schedule.py", line 115, in __singleton_init__
    self.time_offset = self.functions.get('timezone.get_offset', lambda: '0000')()
  File "c:\salt\bin\lib\site-packages\salt\modules\win_timezone.py", line 235, in get_offset
    tz_object = pytz.timezone(get_zone())
  File "c:\salt\bin\lib\site-packages\pytz\__init__.py", line 178, in timezone
    raise UnknownTimeZoneError(zone)
pytz.exceptions.UnknownTimeZoneError: 'Unknown'


I traced this error back as far as I could to see that it was in the get_zone() portion of "tz_object = pytz.timezone(get_zone()). This was returning 'Unknown' because the win_timezone.py module was unable to map the timezone for "Central Standard Time".

This was having issues because that module queries the registry for the TimeZoneKeyName and that had many irrelevant characters following the string. I was able to see this by viewing the binary data under that registry key and then viewing it again after manually putting "Central Standard Time" as the data. I could revert it back to the binary data with the garbage characters by changing the timezone via the Windows clock and then changing it back to Central Standard Time.

It appears that on commit fb0cb23a39f2ab727c0d8b2d5714fc365a72bfdd a developer changed the implementation to use tzutil to return cleaner data but this was 29 days ago and therefore is not in salt minion 2018.3.3.

My company will most likely be getting around this by enforcing that timezone as with a GPO - it seems easier than rebuilding the installer for our Windows workstations. We are unsure of when the next salt minion will be coming out so we don't have any other choice at the moment.

**Comment by twangboy** [comment]
@afischer-opentext-com That should be in 2018.3.4 which should occur within the month. You could patch the module and sync it down to your minions from the master as well. You'd just have to remember to remove it once you upgraded to the version that includes the patched file.

**Comment by twangboy** [comment]
Actually, that got merged into develop. We should back port this to 2018.3.4. I'll do that.

**Comment by twangboy** [comment]
Fixed by #51095

**Comment by twangboy** [comment]
@afischer-opentext-com Thanks for figuring this out. I backported the fix to 2019.2.0 and 2018.3.4

**Comment by afischer-opentext-com** [comment]
Thanks much! Will wait for the new 2018.3.4. We saw the exception did not recognize its impact. We assumed - don't know why - that an exception in an module would have been catched by the calling code/minion process.
